src/native/libs/System.Globalization.Native/pal_locale.m

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_locale_internal.h"
+#import <Foundation/Foundation.h>
+
+const char* DetectDefaultAppleLocaleName()
+{
+    NSLocale *currentLocale = [NSLocale currentLocale];
+    NSString *localeName = @"";
+    
+    if (!currentLocale)
+    {
+        return strdup([localeName UTF8String]);
+    }
+    
+    if ([currentLocale.languageCode length] > 0 && [currentLocale.countryCode length] > 0)
```

Comment:
>currentLocale [](http://example.com/codeflow?start=9&length=13)

do you know if possible having a collation parts too? I mean ICU can have a locale name like:

`de_DE@collation=phoneb` which will be translated to .NET name as `de-DE_phoneb`.

Reply:
It's possible if it grabs the `localeIdentifier` property.  

https://developer.apple.com/documentation/foundation/nslocale/1416263-localeidentifier?language=objc

Reply:
great, I would suggest getting `localeIdentifier` then normalize it to the .NET culture name form.